airbyte-ci/connectors/pipelines/pipelines/bases.py

```diff
@@ -44,6 +54,10 @@ class ConnectorWithModifiedFiles(Connector):
     def has_metadata_change(self) -> bool:
         return any(path.name == METADATA_FILE_NAME for path in self.modified_files)
 
+@dataclass(frozen=True)
+class CDKWithModifiedFiles(CDK):
```

Comment:
I wasn't able to hook into this without further study of the code, so modified_files is empty with all instantiations, but I left the stub here to match the existing implementation